common/util.go

```diff
@@ -2,42 +2,101 @@ package common
 
 import (
 	"fmt"
+	"io/ioutil"
+	"net/http"
 	"os"
 	"strconv"
 	"strings"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
 )
 
-// FissionMetadata contains common fission side fields
-type FissionMetadata struct {
-	// fission
+// ConnectorMetadata contains common fields used by connectors
+type ConnectorMetadata struct {
 	Topic         string
 	ResponseTopic string
 	ErrorTopic    string
-	FunctionURL   string
+	HTTPEndpoint  string
 	MaxRetries    int
 	ContentType   string
-	TriggerName   string
+	SourceName    string
 }
 
-// ParseFissionMetadata parses fission side common fields and returns as fissionMetadata or returns error
-func ParseFissionMetadata() (FissionMetadata, error) {
-	for _, envVars := range []string{"TOPIC", "FUNCTION_URL", "MAX_RETRIES", "CONTENT_TYPE", "TRIGGER_NAME"} {
+// ParseConnectorMetadata parses connector side common fields and returns as ConnectorMetadata or returns error
+func ParseConnectorMetadata() (ConnectorMetadata, error) {
+	for _, envVars := range []string{"TOPIC", "HTTP_ENDPOINT", "MAX_RETRIES", "CONTENT_TYPE"} {
 		if os.Getenv(envVars) == "" {
-			return FissionMetadata{}, fmt.Errorf("environment variable not found: %v", envVars)
+			return ConnectorMetadata{}, fmt.Errorf("environment variable not found: %v", envVars)
 		}
 	}
-	meta := FissionMetadata{
+	meta := ConnectorMetadata{
 		Topic:         os.Getenv("TOPIC"),
 		ResponseTopic: os.Getenv("RESPONSE_TOPIC"),
 		ErrorTopic:    os.Getenv("ERROR_TOPIC"),
-		FunctionURL:   os.Getenv("FUNCTION_URL"),
+		HTTPEndpoint:  os.Getenv("HTTP_ENDPOINT"),
 		ContentType:   os.Getenv("CONTENT_TYPE"),
-		TriggerName:   os.Getenv("TRIGGER_NAME"),
+		SourceName:    os.Getenv("SOURCE_NAME"),
+	}
+	if meta.SourceName == "" {
+		meta.SourceName = "KEDAConnector"
 	}
 	val, err := strconv.ParseInt(strings.TrimSpace(os.Getenv("MAX_RETRIES")), 0, 64)
 	if err != nil {
-		return FissionMetadata{}, fmt.Errorf("failed to parse value from MAX_RETRIES environment variable %v", err)
+		return ConnectorMetadata{}, fmt.Errorf("failed to parse value from MAX_RETRIES environment variable %v", err)
 	}
 	meta.MaxRetries = int(val)
 	return meta, nil
 }
+
+// HandleHTTPRequest sends message and headers data to HTTP endpoint using POST method and returns error in case of failure
+func HandleHTTPRequest(message string, headers map[string]string, data ConnectorMetadata, logger *zap.Logger) error {
```

Comment:
Would you not want to return the body of function response so that it can be sent to the response queue? So return of `(string, error)` maybe?

Reply:
Done